modules/core/keeper/msg_server.go

```diff
@@ -772,7 +772,6 @@ func (k Keeper) ChannelUpgradeInit(goCtx context.Context, msg *channeltypes.MsgC
 		return nil, errorsmod.Wrapf(err, "channel upgrade init callback failed for port ID: %s, channel ID: %s", msg.PortId, msg.ChannelId)
 	}
 
-	upgrade.Fields.Version = upgradeVersion
 	channel := k.ChannelKeeper.WriteUpgradeInitChannel(ctx, msg.PortId, msg.ChannelId, upgrade)
```

Comment:
We should pass `upgradeVersion` to `WriteUpgradeInitChannel` and set `upgrade.Fields.Version` inside that function.

Reply:
I got it, but changing `WriteUpgradeInitChannel ` may affect to  `ChanUpgradeTry`. How do we deal with the case there is no version returned by `OnChanUpgradeInit`